File: authority-portal-backend/authority-portal-api/src/main/java/de/sovity/authorityportal/api/model/CentralComponentCreateRequest.java
package de.sovity.authorityportal.api.model;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Information for registering a new central dataspace component.")
public class CentralComponentCreateRequest {
    @NotBlank(message = "Name of component cannot be blank")
    @Schema(description = "Component Name", requiredMode = Schema.RequiredMode.REQUIRED)
    private String name;

    @Schema(description = "Home Page URL")
    private String homepageUrl;

    @NotBlank(message = "Endpoint URL cannot be blank")
    @Schema(description = "Endpoint URL", requiredMode = Schema.RequiredMode.REQUIRED)
    private String endpointUrl;

    @NotBlank(message = "Certificate cannot be blank")
    @Schema(description = "The component's certificate", requiredMode = Schema.RequiredMode.REQUIRED)
    private String certificate;
}

File: authority-portal-backend/authority-portal-api/src/main/java/de/sovity/authorityportal/api/model/CentralComponentCreateRequest.kt
/*
 * Copyright (c) 2024 sovity GmbH
 *
 * This program and the accompanying materials are made available under the
 * terms of the Apache License, Version 2.0 which is available at
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * SPDX-License-Identifier: Apache-2.0
 *
 * Contributors:
 *      sovity GmbH - initial implementation
 */

package de.sovity.authorityportal.api.model

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotBlank

@Schema(description = "Information for registering a new central dataspace component.")
data class CentralComponentCreateRequest(
    @field:Schema(description = "Component Name", requiredMode = Schema.RequiredMode.REQUIRED)
    @field:NotBlank(message = "Name of component cannot be blank")
    val name: String,

    @field:Schema(description = "Home Page URL", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
    val homepageUrl: String?,

    @field:Schema(description = "Endpoint URL", requiredMode = Schema.RequiredMode.REQUIRED)
    @field:NotBlank(message = "Endpoint URL cannot be blank")
    val endpointUrl:  String,

    @field:Schema(description = "The component's certificate", requiredMode = Schema.RequiredMode.REQUIRED)
    @field:NotBlank(message = "Certificate cannot be blank")
    val certificate: String
)
